feature-libs/product-configurator/rulebased/components/attribute/types/drop-down/configurator-attribute-drop-down.component.ts

```diff
@@ -1,134 +1,38 @@
+  ngOnInit() {
+    this.attributeDropDownForm.setValue(this.attribute?.selectedSingleValue);
+  }
 
-  // TODO(#11681): make quantityService a required dependency
   /**
-   * default constructor
-   * @param {ConfiguratorAttributeQuantityService} quantityService
+   TODO(issue: #11238): update @deprecated level to the release we are publishing with,
+   It is still 3.1 only because app.module.ts states that we are on 3.1.
+   Finally we must have 3.x, x>=2 here
    */
-  // eslint-disable-next-line @typescript-eslint/unified-signatures
-  constructor(quantityService: ConfiguratorAttributeQuantityService);
-
   /**
-   * @deprecated since 3.3
+   * @deprecated since 3.1
+   * User better onSelect(this.attributeDropDownForm.value)
    */
-  constructor();
-
-  constructor(
-    protected quantityService?: ConfiguratorAttributeQuantityService
-  ) {
-    super();
-  }
-
-  ngOnInit() {
-    this.attributeDropDownForm.setValue(this.attribute.selectedSingleValue);
-  }
-
-  get withQuantity() {
-    return (
-      this.quantityService?.withQuantity(
-        this.attribute.dataType,
-        this.attribute.uiType
-      ) ?? false
-    );
-  }
-
-  get disableQuantityActions() {
-    return (
-      this.quantityService?.disableQuantityActions(
-        this.attribute.selectedSingleValue
-      ) ?? true
-    );
-  }
-
   onSelect(): void {
-    this.loading$.next(true);
-
-    const event: ConfigFormUpdateEvent = {
-      changedAttribute: {
-        ...this.attribute,
-        selectedSingleValue: this.attributeDropDownForm.value,
-      },
-      ownerKey: this.ownerKey,
-      updateType: Configurator.UpdateType.ATTRIBUTE,
-    };
-    this.selectionChange.emit(event);
-  }
-
-  onHandleQuantity(quantity: number): void {
-    this.loading$.next(true);
-
-    const event: ConfigFormUpdateEvent = {
-      changedAttribute: {
-        ...this.attribute,
-        quantity,
-      },
-      ownerKey: this.ownerKey,
-      updateType: Configurator.UpdateType.ATTRIBUTE_QUANTITY,
-    };
-
-    this.selectionChange.emit(event);
-  }
-
-  onChangeQuantity(eventObject: any): void {
-    if (!eventObject) {
-      this.attributeDropDownForm.setValue('');
-      this.onSelect();
-    } else {
-      this.onHandleQuantity(eventObject);
-    }
-  }
-
-  /**
-   *  Extract corresponding quantity parameters
-   *
-   * @return {ConfiguratorAttributeQuantityComponentOptions} - New quantity options
-   */
-  extractQuantityParameters(): ConfiguratorAttributeQuantityComponentOptions {
-    const initialQuantity =
-      this.attributeDropDownForm.value !== '0'
-        ? this.attribute.quantity ?? 0
-        : 0;
-
-    return {
-      allowZero: !this.attribute.required,
-      initialQuantity: initialQuantity,
-      disableQuantityActions$: this.loading$.pipe(
-        map((loading) => {
-          return loading || this.disableQuantityActions;
-        })
-      ),
-    };
+    super.onSelect(this.attributeDropDownForm?.value);
```

Comment:
if this function is deprecated, then we have to call `onSelect(attributeDropDownForm.value)` from html.

Reply:
Done!